### PR TITLE
Add --without-cracklib configure parameter

### PR DESCRIPTION
**Pull request by StefanYohansson**
Check for cracklib by default but leave the option to disable it.

closes https://github.com/libpwquality/libpwquality/issues/4

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -153,12 +153,18 @@ fi
 AC_DEFINE_UNQUOTED(CONF_PATH_RANDOMDEV, "$opt_randomdev", [Random device path.])
 
 dnl Check for cracklib
-AC_CHECK_HEADERS([crack.h],
-      AC_SEARCH_LIBS([FascistCheck], [crack], LIBCRACK="-lz -lcrack" LIBS="$LIBS -lz", LIBCRACK="", [-lz]))
-if test "x$LIBCRACK" = "x" ; then
-    AC_MSG_ERROR([No or unusable cracklib library])
+AC_ARG_WITH([cracklib], AS_HELP_STRING([--without-cracklib],[disable cracklib]), [], [opt_cracklib=yes])
```
Comment (t8m) [inline configure.ac:156]: Please change this to AC_ARG_ENABLE and the description should be `disable cracklib dictionary check` or something like that.

File: configure.ac [hunk 2]
```diff
@@ -153,12 +153,18 @@ fi
...
+if test x"$opt_cracklib" = xyes; then
+       AC_CHECK_HEADERS([crack.h],
+              AC_SEARCH_LIBS([FascistCheck], [crack], LIBCRACK="-lz -lcrack" LIBS="$LIBS -lz", LIBCRACK="", [-lz]))
+              if test "x$LIBCRACK" = "x" ; then
+                    AC_MSG_ERROR([No or unusable cracklib library])
+              fi
+       AC_SUBST([LIBCRACK])
+       AC_DEFINE([USE_CRACKLIB], [1], [define usage of cracklib])
```
Comment (t8m) [inline configure.ac:164]: You can drop the USE_CRACKLIB defines in both if and else branches.

File: src/check.c [hunk 3]
```diff
@@ -9,7 +9,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
+#ifndef USE_CRACKLIB
```
Comment (t8m) [inline src/check.c:12]: And then everywhere replace `ifndef USE_CRACKLIB` (which is wrong anyway) with `ifdef HAVE_CRACK_H`.

File: src/settings.c [hunk 4]
```diff
@@ -476,10 +478,12 @@ pwquality_get_str_value(pwquality_settings_t *pwq, int setting, const char **val
                 *value = pwq->bad_words;
                 break;
         case PWQ_SETTING_DICT_PATH:
+                #ifndef USE_CRACKLIB
                 if (pwq->dict_path)
                         *value = pwq->dict_path;
                 else
                         *value = GetDefaultCracklibDict();
+                #endif
```
Comment (t8m) [inline src/settings.c:486]: please return NULL as the value if the cracklib is disabled.

**Review by t8m — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (StefanYohansson) [reply configure.ac:156]: Thank you for the suggestion, I will change it.

Sorry for the mistake, I am not proficient with Autoconf and used what I already seem on this file.

Comment (StefanYohansson) [reply configure.ac:156]: @t8m any problem the parameter should be `--disable-cracklib` ?

Comment (t8m) [reply configure.ac]: Hmm, maybe `--disable-cracklib-check` so the enable should be for cracklib-check.

File: configure.ac [hunk 5]
```diff
@@ -153,12 +153,17 @@ fi
 AC_DEFINE_UNQUOTED(CONF_PATH_RANDOMDEV, "$opt_randomdev", [Random device path.])
 
 dnl Check for cracklib
+AC_ARG_ENABLE([cracklib],
+        AS_HELP_STRING([--disable-cracklib], [disable cracklib dictionary check]))
```
Comment (t8m) [inline configure.ac:157]: As I've replied, you have a good point that it should be probably cracklib-check and --disable-cracklib-check.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged, thank you for the PR.

**Pull request merged**